### PR TITLE
Only show search context controls if the current context matches the …

### PR DESCRIPTION
**Pull request by cbeer**
…current document

Maybe helps address or obscure the problems from #2295?

### CODE REVIEW

File: app/views/catalog/_show_main_content.html.erb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<%= render 'previous_next_doc' if @search_context %>
+<%= render 'previous_next_doc' if @search_context && @search_context['document_id'] == @document.id %>
```
Comment (cdmo) [inline app/views/catalog/_show_main_content.html.erb:1]: Is this right?

```suggestion
<%= render 'previous_next_doc' if @search_context && search_session['document_id'] == @document.id %>
```

**Comment by cdmo** [comment]
I just tested this fix out locally and it worked well once I made a small change. I personally consider this a useful and important fix. Thanks. Happy to help flesh this out more if needed.

File: app/helpers/blacklight/url_helper_behavior.rb [hunk 2]
```diff
@@ -68,7 +68,7 @@ def link_to_next_document(next_document)
   #   session_tracking_params(SolrDocument.new(id: 123), 7)
   #   => { data: { :'context-href' => '/catalog/123/track?counter=7&search_id=999' } }
   def session_tracking_params document, counter
-    path = session_tracking_path(document, per_page: params.fetch(:per_page, search_session['per_page']), counter: counter, search_id: current_search_session.try(:id))
+    path = session_tracking_path(document, per_page: params.fetch(:per_page, search_session['per_page']), counter: counter, search_id: current_search_session.try(:id), document_id: document.id)
```
Comment (cdmo) [inline app/helpers/blacklight/url_helper_behavior.rb:71]: Without a safe navigation operator you get a `id` not defined on nil error when you click on the first result.

```suggestion
    path = session_tracking_path(document, per_page: params.fetch(:per_page, search_session['per_page']), counter: counter, search_id: current_search_session.try(:id), document_id: document&.id)
```

**Comment by cdmo** [comment]
I am on 7.5.1 for my purposes and we have some local differences in our placement of where we render `previous_next_doc`, but so far in testing things out locally this PR is working!

**Comment by cdmo** [comment]
Just rebased and then added a rubocop fix here.

**Comment by cdmo** [comment]
I took the liberty of fleshing this PR out some more, hope that's okay. I wasn't sure of a couple of thing: 1) if the `url_helper_behavior` needed additional testing and if `views/catalog/_show.html.erb_spec.rb:1` needed a corresponding view spec - I didn't see one at all for this partial.

**Review by mejackreed — APPROVED** [review]

**Pull request merged**